pkg/infrared/server.go

```diff
@@ -257,42 +253,32 @@ func (s *statusResponseProvider) StatusResponse(protVer protocol.Version, readPk
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	hash, ok := s.statusHash[protVer]
-	if !ok {
-		hash, newStatusResp, pk, err := s.requestNewStatusResponseJSON(readPks)
-		if err != nil {
-			return status.ResponseJSON{}, protocol.Packet{}, err
-		}
-		s.statusHash[protVer] = hash
-
-		entry, ok := s.statusResponseCache[hash]
-		if !ok {
-			s.statusResponseCache[hash] = &statusCacheEntry{
-				expiresAt:    time.Now().Add(s.cacheTTL),
-				responseJSON: newStatusResp,
-				responsePk:   pk,
-			}
-			return newStatusResp, pk, nil
-		}
-		return entry.responseJSON, entry.responsePk, nil
+	hash, okHash := s.statusHash[protVer]
+	entry, okCache := s.statusResponseCache[hash]
+	if !okHash || !okCache {
+		return s.CacheResponse(protVer, readPks)
 	}
 
-	entry, ok := s.statusResponseCache[hash]
-	if !ok {
-		hash, newStatusResp, pk, err := s.requestNewStatusResponseJSON(readPks)
-		if err != nil {
-			return status.ResponseJSON{}, protocol.Packet{}, err
-		}
+	return entry.responseJSON, entry.responsePk, nil
+}
 
-		s.statusResponseCache[hash] = &statusCacheEntry{
-			expiresAt:    time.Now().Add(s.cacheTTL),
-			responseJSON: newStatusResp,
-			responsePk:   pk,
-		}
+func (s *statusResponseProvider) CacheResponse(protVer protocol.Version, readPks [2]protocol.Packet) (status.ResponseJSON, protocol.Packet, error) {
+	newStatusResp, pk, err := s.requestNewStatusResponseJSON(readPks)
+	if err != nil {
+		return status.ResponseJSON{}, protocol.Packet{}, err
+	}
 
-		return newStatusResp, pk, nil
+	hash := xxhash.New()
+	pk.WriteTo(hash)
+	s.statusHash[protVer] = hash.Sum64()
+
+	s.statusResponseCache[hash.Sum64()] = &statusCacheEntry{
```

Comment:
I don't like how hash.Sum64() is called twice and especially the second time makes it kinda hard to read since there is already so much going on. It might be better to create a separate variable like `hashSum`.

Reply:
Alright, lets fix that